diesel_codegen/src/schema_inference/data_structures.rs

```diff
@@ -1,54 +1,45 @@
 use diesel::*;
+#[cfg(feature = "postgres")]
 use diesel::pg::Pg;
+#[cfg(feature = "sqlite")]
+use diesel::sqlite::Sqlite;
 use diesel::types::{HasSqlType, FromSqlRow};
 
-table! {
-    pg_attribute (attrelid) {
-        attrelid -> Oid,
-        attname -> VarChar,
-        atttypid -> Oid,
-        attnotnull -> Bool,
-        attnum -> SmallInt,
-        attisdropped -> Bool,
-    }
-}
-
-table! {
-    pg_type (oid) {
-        oid -> Oid,
-        typname -> VarChar,
-    }
-}
-
-joinable!(pg_attribute -> pg_type (atttypid));
-select_column_workaround!(pg_attribute -> pg_type (attrelid, attname, atttypid, attnotnull, attnum, attisdropped));
-select_column_workaround!(pg_type -> pg_attribute (oid, typname));
-
-table! {
-    pg_class (oid) {
-        oid -> Oid,
-        relname -> VarChar,
-    }
-}
-
 #[derive(Debug, Clone)]
-pub struct PgAttr {
+pub struct ColumnInformation {
     pub column_name: String,
     pub type_name: String,
     pub nullable: bool,
 }
 
-impl<ST> Queryable<ST, Pg> for PgAttr where
+#[cfg(feature = "postgres")]
+impl<ST> Queryable<ST, Pg> for ColumnInformation where
     Pg: HasSqlType<ST>,
     (String, String, bool): FromSqlRow<ST, Pg>,
 {
     type Row = (String, String, bool);
 
     fn build(row: Self::Row) -> Self {
-        PgAttr {
+        ColumnInformation {
             column_name: row.0,
             type_name: row.1,
             nullable: !row.2,
         }
     }
 }
+
+#[cfg(feature = "sqlite")]
+impl<ST> Queryable<ST, Sqlite> for ColumnInformation where
+    Sqlite: HasSqlType<ST>,
+    (i32, String, String, i32, Option<String>, i32): FromSqlRow<ST, Sqlite>,
+{
+    type Row = (i32, String, String, i32, Option<String>, i32);
+
+    fn build(row: Self::Row) -> Self {
+        ColumnInformation {
+            column_name: row.1,
+            type_name: row.2,
+            nullable: row.3 == 0,
```

Comment:
If I try to use a bool here the compiler is complaining that `the trait diesel::types::FromSqlRow<diesel::types::Integer, diesel::sqlite::backend::Sqlite> is not implemented for the type bool`